test/bigtest/tests/units/settings/MappingProfilesForm/TransformationsModal-test.js

```diff
@@ -148,30 +152,136 @@ describe('MappingProfilesTransformationsModal', () => {
-        expect(modal.transformations.list.rowCount).to.equal(1);
-        expect(modal.resultsPane.header.sub).to.equal('1 field found');
+      it('should select all transformations', () => {
+        expect(modal.transformations.checkboxes(0).isChecked).to.be.true;
+        expect(modal.transformations.checkboxes(1).isChecked).to.be.true;
+        expect(modal.transformations.checkboxes(2).isChecked).to.be.true;
+      });
+
+      it('should check select all checkboxes', () => {
+        expect(modal.transformations.selectAllCheckbox.isChecked).to.be.true;
+      });
+
+      it('should update total selected count', () => {
+        expect(modal.totalSelected.text).to.equal(totalSelectedMessage(3));
+      });
+
+      describe('clicking on checkbox from the first row', () => {
+        beforeEach(async () => {
+          await modal.transformations.checkboxes(0).clickInput();
+        });
+
+        it('should deselect the first item', () => {
+          expect(modal.transformations.checkboxes(0).isChecked).to.be.false;
+        });
+
+        it('should uncheck select all checkboxes', () => {
+          expect(modal.transformations.selectAllCheckbox.isChecked).to.be.false;
+        });
+
+        it('should update total selected count', () => {
+          expect(modal.totalSelected.text).to.equal(totalSelectedMessage(2));
+        });
+
+        describe('clicking on select all checkboxes', () => {
+          beforeEach(async () => {
+            await modal.transformations.selectAllCheckbox.clickInput();
+          });
+
+          it('should select all transformations (apply selection to unselected items)', () => {
+            expect(modal.transformations.checkboxes(0).isChecked).to.be.true;
+            expect(modal.transformations.checkboxes(1).isChecked).to.be.true;
+            expect(modal.transformations.checkboxes(2).isChecked).to.be.true;
+          });
+
+          it('should check select all checkboxes', () => {
+            expect(modal.transformations.selectAllCheckbox.isChecked).to.be.true;
+          });
+
+          it('should update total selected count', () => {
+            expect(modal.totalSelected.text).to.equal(totalSelectedMessage(3));
+          });
+        });
       });
 
-      it('should filter out the transformation list', () => {
-        expect(modal.transformations.list.rows(0).cells(1).text).to.equal(initialValues.transformations[1].displayName);
+      describe('clicking on select all checkboxes', () => {
+        beforeEach(async () => {
+          await modal.transformations.selectAllCheckbox.clickInput();
+        });
+
+        it('should deselect all transformations', () => {
+          expect(modal.transformations.checkboxes(0).isChecked).to.be.false;
+          expect(modal.transformations.checkboxes(1).isChecked).to.be.false;
+          expect(modal.transformations.checkboxes(2).isChecked).to.be.false;
+        });
+
+        it('should uncheck select all checkboxes', () => {
```

Comment:
```suggestion
        it('should uncheck select all checkbox', () => {
```

Similar [here](https://github.com/folio-org/ui-data-export/pull/122/files#diff-491c106836b32bfb2aaaf5a44c7de2f4R212) and [here](https://github.com/folio-org/ui-data-export/pull/122/files#diff-491c106836b32bfb2aaaf5a44c7de2f4R176).